corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js

```diff
@@ -343,20 +343,35 @@ hqDefine("cloudcare/js/formplayer/app", function () {
                 false
             );
         }
+
+        const pendingErrors = [];
+
         window.addEventListener(
-            'offline', function () {
-                showError(gettext("You are now offline. Web Apps is not optimized " +
-                    "for offline use. Please reconnect to the Internet before " +
-                    "continuing."), $("#cloudcare-notifications"));
-                $('.submit').prop('disabled', 'disabled');
-                $('.form-control').prop('disabled', 'disabled');
-            }
-        );
+            'offline',function () {
+                const offlineHandler = () => {
+                    showError(gettext("You are now offline. Web Apps is not optimized " +
+                        "for offline use. Please reconnect to the Internet before " +
+                        "continuing."), $("#cloudcare-notifications"));
+                    $('.submit').prop('disabled', 'disabled');
+                    $('.form-control').prop('disabled', 'disabled');
+                    pendingErrors.pop();
+                };
+                if (pendingErrors.length) {
+                    offlineHandler();
+                } else {
+                    pendingErrors.push(setTimeout(offlineHandler,2000));
```

Comment:
Avoid magic numbers like 2000. Create a constant for what 2000 represents, in this case the reconnect_timing_window or something. It's possible in the future that 2 seconds needs to become 5 seconds, and that change is much easier if this value is stored in a self-documenting way